azkaban-common/src/main/java/azkaban/executor/FetchActiveFlowDao.java

```diff
@@ -124,6 +128,20 @@ Map<Integer, Pair<ExecutionReference, ExecutableFlow>> fetchUnfinishedFlows()
     }
   }
 
+  Pair<ExecutionReference, ExecutableFlow> fetchUnfinishedFlow(final int executionId)
+      throws ExecutorManagerException {
+    try {
+      Iterator<Pair<ExecutionReference, ExecutableFlow>> iterator = this.dbOperator.query(FetchActiveExecutableFlows.FETCH_UNFINISHED_EXECUTABLE_FLOW_BY_EXECID,
+          new FetchActiveExecutableFlows(), executionId).values().iterator();
+      if (iterator.hasNext()) {
+        return iterator.next();
+      }
+      return null;
```

Comment:
If we're returning null, can we make this into an optional?

Reply:
Yeah, I also thought about that. After looking into our codebase, returning null is already like a convention for us:
1. https://github.com/azkaban/azkaban/blob/bfd36e24835f292a38b2afac2451580f06637267/azkaban-common/src/main/java/azkaban/executor/FetchActiveFlowDao.java#L297
2. https://github.com/azkaban/azkaban/blob/dc7203d512cd03172b8f85a4012aae488c387f09/azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java#L209